crates/sparrow-compiler/src/ast_to_dfg.rs

```diff
@@ -647,6 +647,16 @@ fn cast_if_needed(
         // TODO: This maybe should create an error node and return it.
         (_, FenlType::Error) => Ok(value),
         (actual, expected) if actual == expected => Ok(value),
+
+        // This handles the case where user `map` types have different names but are
+        // otherwise identical. When constructing the concrete map during inference,
+        // we make up names for the map types, so this is a common case. This ensures that
+        // the types themselves are still treated as equal.
+        (FenlType::Concrete(DataType::Map(s, _)), FenlType::Concrete(DataType::Map(s2, _)))
```

Comment:
Why do map types have names at all? Is this they have different *field names* in the struct?

Reply:
Updated language to say _field_ names